src/api/article/data/index.js

```diff
@@ -0,0 +1,14 @@
+import { Router } from 'express';
+
+import { requireAuth, verifyPermission } from 'auth';
+
+import * as controller from './controller';
+import ArticleData from './model';
+
+const router = Router();
+
+router.post('/:articleId', requireAuth, verifyPermission('canCreateArticle'), controller.create);
+router.put('/:slug', requireAuth, verifyPermission('canCreateArticle'), controller.update);
```

Comment:
What about using `router.put('/:articleId/:slug'`? Thus, it is more clearly that create and put are applied to different entities. (usually `post on /` - create, `put on /:id` - edit)

Reply:
Lol, how it can work at all? `controller.update` will never be called!

Reply:
It will be if http query is `PUT` rather than `POST`.

`/:articleId/:slug` seems to have redundant information. We'll need additional checks if the `slug` is really a part of the `articleId`.